File: app/src/main/java/com/nikola/jakshic/truesight/view/PeerSortDialog.java
package com.nikola.jakshic.truesight.view;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.nikola.jakshic.truesight.R;

public class PeerSortDialog extends DialogFragment {

    public interface OnSortListener {
        void onSort(int sortOption);
    }

    public static PeerSortDialog newInstance() {
        PeerSortDialog dialog = new PeerSortDialog();
        return dialog;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setTitle("Sort by")
                .setItems(R.array.sort_peer_options, (dialog, which) ->
                        ((OnSortListener) getTargetFragment()).onSort(which))
                .create();
    }
}

File: app/src/main/java/com/nikola/jakshic/dagger/ui/profile/peer/PeerSortDialog.kt
package com.nikola.jakshic.dagger.ui.profile.peer

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import com.nikola.jakshic.dagger.R

class PeerSortDialog : DialogFragment() {

    interface OnSortListener {
        fun onSort(sort: Int)
    }

    companion object {
        fun newInstance() = PeerSortDialog()
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        return AlertDialog.Builder(context!!)
                .setTitle("Sort by")
                .setItems(R.array.sort_peer_options) { _, which ->
                    (targetFragment as OnSortListener).onSort(which)
                }
                .create()
    }
}